modules/imonomyBidAdapter.js

```diff
@@ -0,0 +1,126 @@
+import * as utils from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'imonomy';
+const ENDPOINT = '//b.imonomy.com/openrtb/hb/00000';
+const USYNCURL = '//b.imonomy.com/UserMatching/b/';
+
+export const spec = {
+  code: BIDDER_CODE,
+
+  /**
+  * Determines whether or not the given bid request is valid. Valid bid request must have placementId and hbid
+  *
+  * @param {BidRequest} bid The bid params to validate.
+  * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  isBidRequestValid: bid => {
+    return !!(bid && bid.params && bid.params.placementId && bid.params.hbid);
+  },
+  /**
+  * Make a server request from the list of BidRequests.
+  *
+  * @param {validBidRequests[]} - an array of bids
+  * @return ServerRequest Info describing the request to the server.
+  */
+  buildRequests: validBidRequests => {
+    const tags = validBidRequests.map(bid => {
+      // map each bid id to bid object to retrieve adUnit code in callback
+      let tag = {
+        uuid: bid.bidId,
+        sizes: bid.sizes,
+        trid: bid.transactionId,
+        hbid: bid.params.hbid,
+        placementid: bid.params.placementId
+      };
+
+      // add floor price if specified (not mandatory)
+      if (bid.params.floorPrice) {
+        tag.floorprice = bid.params.floorPrice;
+      }
+
+      return tag;
+    });
+
+    // Imonomy server config
+    const time = new Date().getTime();
+    const kbConf = {
+      ts_as: time,
+      hb_placements: [],
+      hb_placement_bidids: {},
+      hb_floors: {},
+      cb: _generateCb(time),
+      tz: new Date().getTimezoneOffset(),
+    };
+
+    validBidRequests.forEach(bid => {
+      kbConf.hdbdid = kbConf.hdbdid || bid.params.hbid;
+      kbConf.encode_bid = kbConf.encode_bid || bid.params.encode_bid;
+      kbConf.hb_placement_bidids[bid.params.placementId] = bid.bidId;
```

Comment:
Currently on line 58, the `kbConf.encode_bid` value is always going to be `undefined`, since it is not declared/ assigned anywhere before this point.

So, change line 58 to something like: 
`kbConf.encode_bid = bid.params.encode_bid;`